src/Sylius/Behat/Context/Setup/CustomerGroupContext.php

```diff
@@ -54,22 +54,36 @@ public function __construct(
 
     /**
      * @Given the store has a customer group :name
+     * @Given the store has a customer group :name with :code code
      */
-    public function theStoreHasACustomerGroup($name)
+    public function theStoreHasACustomerGroup($name, $code = null)
     {
-        $this->createCustomerGroup($name);
+        $this->createCustomerGroup($name, $code);
     }
 
     /**
      * @param string $name
+     * @param string $code
      */
-    private function createCustomerGroup($name)
+    private function createCustomerGroup($name, $code)
     {
         /** @var CustomerGroupInterface $customerGroup */
         $customerGroup = $this->customerGroupFactory->createNew();
+
+        $customerGroup->setCode($code ? $code : $this->generateCodeFromName($name));
         $customerGroup->setName(ucfirst($name));
 
         $this->sharedStorage->set('customer_group', $customerGroup);
         $this->customerGroupRepository->add($customerGroup);
     }
+
+    /**
+     * @param string $name
+     *
+     * @return string
+     */
+    private function generateCodeFromName($name)
+    {
+        return str_replace(' ', '_', strtoupper($name));
```

Comment:
I think we can use ``Sylius\Component\Core\Formatter\StringInflector::nameToUppercaseCode($name)``.